tests/Opus/Search/Common/IndexerTest.php

```diff
@@ -0,0 +1,961 @@
+	 */
+	public function testMissingConfigParamSearchEngine_Index_Host() {
+		$this->markTestSkipped( 'not supported in Opus_Search API due to passing configuration to 3rd-party library for processing' );
+
+		$this->adjustConfiguration( array(), function ( Zend_Config $config ) {
+			unset(
+				$config->searchengine->solr->default->service->index->endpoint->primary->host,
+				$config->searchengine->solr->default->service->default->endpoint->primary->host
+			);
+
+			return $config;
+		} );
+
+		Opus_Search_Service::selectIndexingService()
+		                   ->removeAllDocumentsFromIndex();
+	}
+
+	/**
+	 * @expectedException Opus_Search_InvalidConfigurationException
+	 */
+	public function testMissingConfigParamSearchEngine_Index_Port() {
+		$this->markTestSkipped( 'not supported in Opus_Search API due to passing configuration to 3rd-party library for processing' );
+
+		$this->adjustConfiguration( array(), function ( Zend_Config $config ) {
+			unset(
+				$config->searchengine->solr->default->service->index->endpoint->primary->port,
+				$config->searchengine->solr->default->service->default->endpoint->primary->port
+			);
+
+			return $config;
+		} );
+
+		Opus_Search_Service::selectIndexingService()
+		                   ->removeAllDocumentsFromIndex();
+	}
+
+	/**
+	 * @expectedException Opus_Search_InvalidConfigurationException
+	 */
+	public function testMissingConfigParamSearchEngine_Index_Path() {
+		$this->markTestSkipped( 'not supported in Opus_Search API due to passing configuration to 3rd-party library for processing' );
+
+		$this->adjustConfiguration( array(), function ( Zend_Config $config ) {
+			unset(
+				$config->searchengine->solr->default->service->index->endpoint->primary->path,
+				$config->searchengine->solr->default->service->default->endpoint->primary->path
+			);
+
+			return $config;
+		} );
+
+		Opus_Search_Service::selectIndexingService()
+		                   ->removeAllDocumentsFromIndex();
+	}
+
+	public function testMissingConfigParamLogPrepareXml() {
+		$this->adjustConfiguration( array(), function ( Zend_Config $config ) {
+			unset( $config->log->prepare->xml );
+
+			return $config;
+		} );
+
+		$this->_addOneDocumentToIndex();
+	}
+
+	/**
+	 * @expectedException Opus_Search_InvalidConfigurationException
+	 */
+	public function testEmptyConfiguration() {
+		$this->markTestSkipped( 'not supported in Opus_Search API due to passing configuration to 3rd-party library for processing' );
+
+		$this->adjustConfiguration( array(
+            'searchengine' => array( 'solr' => array( 'default' => array( 'service' => array( 'index' => array( 'endpoint' => array( 'primary' => array( 'path' => '' ) ) ) ) ) ) )
+        ) );
+
+		Opus_Search_Service::selectIndexingService()
+		                   ->removeAllDocumentsFromIndex();
+	}
+
+	/**
+	 * @expectedException Opus_Search_InvalidConfigurationException
+	 */
+	public function testInvalidConfiguration() {
```

Comment:
Well, basically there are tests on Opus_Search_Service assessing proper indication of failing to access service. These don't assess application though on properly handling those exceptions currently. According to defining API of a package such as Opus_Search returning on error should be considered best approach and so do consider some tests in more specific suites such as Opus_Search_Solr_Solarium_AdapterIndexingTest::testDisfunctServiceFails() in expecting Opus_Search_Exception (due to timeout, probably). May it might be preferred to test those cases in a more general test suite of framework, though.

According to testing application, do you know of any tests assessing this fail-safe behaviour, currently, to be adopted?